llvm/docs/LangRef.rst

```diff
@@ -15434,6 +15434,66 @@ The behavior of '``llvm.memset.inline.*``' is equivalent to the behavior of
 '``llvm.memset.*``', but the generated code is guaranteed not to call any
 external functions.
 
+.. _int_experimental_memset_pattern:
+
+'``llvm.experimental.memset.pattern``' Intrinsic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Syntax:
+"""""""
+
+This is an overloaded intrinsic. You can use
+``llvm.experimental.memset.pattern`` on any integer bit width that is an
+integral number of bytes and for different address spaces. Not all targets
```

Comment:
Do we need a Verifier check to check that the size is integral number of bytes?

Reply:
I've now added this check to the verifier.

Reply:
Actually that was my bad. There was some discussion about this before and in the end non-integral bitwidths should work and are tested in `llvm/test/Transforms/PreISelIntrinsicLowering/RISCV/memset-pattern.ll`. I've now pushed a fix to LangRef to remove the stated restriction.